news-JS/src/components/view/sources/sources.ts

```diff
@@ -0,0 +1,36 @@
+import { sourceObject } from '../../controller/loader';
+import { nonNullQuerySelector } from '../news/news';
+import './sources.css';
+
+class Sources {
+  public draw(
+    data: Readonly<Pick<sourceObject, 'id' | 'name' | 'description' | 'url' | 'category' | 'language' | 'country'>>[]
+  ) {
+    const fragment = document.createDocumentFragment();
+    const sourceItemTemp = document.querySelector('#sourceItemTemp');
+    if (!(sourceItemTemp instanceof HTMLTemplateElement)) {
+      throw new Error('No template element!');
```

Comment:
This is not the best way to avoid a type error because throw error can break the program and the user will not see anything. When we somewhere throw error, after in other place we need catch it and show error message to user